easybuild/easyconfigs/p/Python/Python-3.9.5-GCCcore-10.3.0.eb

```diff
@@ -0,0 +1,441 @@
+name = 'Python'
+version = '3.9.5'
+
+homepage = 'https://python.org/'
+description = """Python is a programming language that lets you work more quickly and integrate your systems
+ more effectively."""
+
+toolchain = {'name': 'GCCcore', 'version': '10.3.0'}
+toolchainopts = {'pic': True}
+
+source_urls = ['https://www.python.org/ftp/%(namelower)s/%(version)s/']
+sources = [SOURCE_TGZ]
+checksums = ['e0fbd5b6e1ee242524430dee3c91baf4cbbaba4a72dd1674b90fda87b713c7ab']
+
+builddependencies = [
+    ('UnZip', '6.0'),
+]
+
+dependencies = [
+    ('binutils', '2.36.1'),
+    ('bzip2', '1.0.8'),  # required for bz2 package in Python stdlib
+    ('zlib', '1.2.11'),
+    ('libreadline', '8.1'),
+    ('ncurses', '6.2'),
+    ('SQLite', '3.35.4'),
+    ('XZ', '5.2.5'),
+    ('GMP', '6.2.1'),  # required for pycrypto
+    ('libffi', '3.3'),  # required for cryptography
+    # OS dependency should be preferred if the os version is more recent then this version,
+    # it's nice to have an up to date openssl for security reasons
+    # Python 3.7 requires OpenSSL >= 1.0.2, uncomment line below if OS version is too old (e.g. CentOS 6)
+    # ('OpenSSL', '1.1.1k'),
+]
+
+osdependencies = [OS_PKG_OPENSSL_DEV]
+
+# symlink 'pip' command to 'pip3' that is included with Python installation
+# required so we can update pip to version included in extensions, using pip
+installopts = " && ln -s %(installdir)s/bin/pip3 %(installdir)s/bin/pip"
+
+exts_default_options = {
+    'download_dep_fail': True,
+    'sanity_pip_check': True,
+    'source_urls': [PYPI_SOURCE],
+    'use_pip': True,
+}
+
+# order is important!
+# package versions updated 10 May 2021
+exts_list = [
+    ('pip', '21.1.1', {
+        'use_pip': False,
+        'checksums': ['51ad01ddcd8de923533b01a870e7b987c2eb4d83b50b89e1bf102723ff9fed8b'],
+    }),
+    ('wheel', '0.36.2', {
+        'checksums': ['e11eefd162658ea59a60a0f6c7d493a7190ea4b9a85e335b33489d9f17e0245e'],
+    }),
+    ('setuptools', '56.2.0', {
+        'source_tmpl': '%(name)s-%(version)s.tar.gz',
+        'checksums': ['7bb5652625e94e73b9358b7ed8c6431b732e80cf31f4e0972294c64f0e5b849e'],
+    }),
+    ('nose', '1.3.7', {
+        'checksums': ['f1bffef9cbc82628f6e7d7b40d7e255aefaa1adb6a1b1d26c69a8b79e6208a98'],
+    }),
+    ('blist', '1.3.6', {
+        'modulename': False,
```

Comment:
> blist is failing without modulename: False.

How is it failing? It looks like something might be wrong then. Do you still have the error message?

Reply:
```
== 2021-05-12 12:09:34,583 build_log.py:169 ERROR EasyBuild crashed with an error (at easybuild/base/exceptions.py:124 in __init__): Sanity check failed: extensions sanity check failed for 1 extensions: blist
failing sanity check for 'blist' extension: command "python -c "import blist"" failed; output:
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/p/usersoftware/slai/jurecadc/stages/2021a/software/Python/3.9.5-GCCcore-10.3.0/lib/python3.9/site-packages/blist/__init__.py", line 2, in <module>
    from blist._blist import *
ImportError: /p/usersoftware/slai/jurecadc/stages/2021a/software/Python/3.9.5-GCCcore-10.3.0/lib/python3.9/site-packages/blist/_blist.cpython-39-x86_64-linux-gnu.so: undefined symbol: _PyObject_GC_IS_TRACKED,  (at easybuild/framework/easyblock.py:2872 in _sanity_check_step)
```
This is a bug of `blist` with Python >= 3.9:
https://bugzilla.redhat.com/show_bug.cgi?id=1830801
https://github.com/DanielStutzbach/blist/issues/90

Unfortunately the blist upstream repo has been inactive for years. Do we want to remove the package? Otherwise there is a patch for `blist` to work with Python >=3.9:
https://github.com/conda-forge/blist-feedstock/blob/master/recipe/patches/0004-compatibility-with-PEP-620.patch